renderdoc/driver/vrapi/vrapi_hooks.cpp

```diff
@@ -0,0 +1,313 @@
+
+  vrapi_hooks.vrapi_LeaveVrMode_real(ovr);
+}
+
+__attribute__((visibility("default"))) ovrTextureSwapChain *vrapi_CreateTextureSwapChain2(
+    ovrTextureType type, ovrTextureFormat format, int width, int height, int levels, int bufferCount)
+{
+  if(vrapi_hooks.vrapi_CreateTextureSwapChain2_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainHandle_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainLength_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  ovrTextureSwapChain *texture_swapchain = vrapi_hooks.vrapi_CreateTextureSwapChain2_real(
+      type, format, width, height, levels, bufferCount);
+
+  int tex_count = vrapi_hooks.vrapi_GetTextureSwapChainLength_real(texture_swapchain);
+
+  if(m_GLDriver)
+  {
+    for(int i = 0; i < tex_count; ++i)
+    {
+      GLuint tex = vrapi_hooks.vrapi_GetTextureSwapChainHandle_real(texture_swapchain, i);
+      GLenum internalformat = GetInternalFormat(format);
+      TextureCategory creationFlags = GetTextureCategory(format);
+
+      m_GLDriver->CreateVRAPITextureSwapChain(tex, internalformat, width, height, creationFlags);
+    }
+  }
+
+  return texture_swapchain;
+}
+
+__attribute__((visibility("default"))) ovrTextureSwapChain *vrapi_CreateTextureSwapChain(
+    ovrTextureType type, ovrTextureFormat format, int width, int height, int levels, bool buffered)
+{
+  if(vrapi_hooks.vrapi_CreateTextureSwapChain_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainHandle_real == NULL ||
+     vrapi_hooks.vrapi_GetTextureSwapChainLength_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  ovrTextureSwapChain *texture_swapchain =
+      vrapi_hooks.vrapi_CreateTextureSwapChain_real(type, format, width, height, levels, buffered);
+
+  int tex_count = vrapi_hooks.vrapi_GetTextureSwapChainLength_real(texture_swapchain);
+
+  for(int i = 0; i < tex_count; ++i)
+  {
+    GLuint tex = vrapi_hooks.vrapi_GetTextureSwapChainHandle_real(texture_swapchain, i);
+    GLenum internalformat = GetInternalFormat(format);
+    TextureCategory creationFlags = GetTextureCategory(format);
+
+    m_GLDriver->CreateVRAPITextureSwapChain(tex, internalformat, width, height, creationFlags);
+  }
+
+  return texture_swapchain;
+}
+
+__attribute__((visibility("default"))) void vrapi_SubmitFrame(ovrMobile *ovr,
+                                                              const ovrFrameParms *parms)
+{
+  if(vrapi_hooks.vrapi_SubmitFrame_real == NULL || vrapi_hooks.vrapi_GetSystemPropertyInt_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  if(m_GLDriver)
+  {
+    SCOPED_LOCK(glLock);
+
+    int width = vrapi_hooks.vrapi_GetSystemPropertyInt_real(&parms->Java,
+                                                            VRAPI_SYS_PROP_DISPLAY_PIXELS_WIDE);
+    int height = vrapi_hooks.vrapi_GetSystemPropertyInt_real(&parms->Java,
+                                                             VRAPI_SYS_PROP_DISPLAY_PIXELS_HIGH);
+
+    m_GLDriver->WindowSize(ovr, width, height);
```

Comment:
But without the window size, I get a 8x8 thumbnail for the capture. Perhaps I can call ANativeWindow_getWidth and ANativeWindow_getHeight to get the real window size?